packages/portfolio-contract/test/rebalance.test.ts

```diff
@@ -0,0 +1,337 @@
+    { src: '@noble', dest: '@Arbitrum', amount: token(100n) },
+    { src: '@Arbitrum', dest: A, amount: token(100n) },
+  ]);
+});
+
+test('solver deposit redistribution (+agoric 100 -> A 70, B 30)', async t => {
+  const current = balances({ '+agoric': 100n, [A]: 0n, [B]: 0n });
+  const { steps } = await planRebalanceFlow({
+    mode: 'cheapest',
+    network: TEST_NETWORK,
+    current,
+    target: { '+agoric': ZERO, [A]: token(70n), [B]: token(30n) },
+    brand: TOK_BRAND,
+  });
+  t.deepEqual(steps, [
+    { src: '+agoric', dest: '@agoric', amount: token(100n) },
+    { src: '@agoric', dest: '@noble', amount: token(100n) },
+    { src: '@noble', dest: '@Arbitrum', amount: token(70n) },
+    { src: '@noble', dest: '@Avalanche', amount: token(30n) },
+    { src: '@Arbitrum', dest: A, amount: token(70n) },
+    { src: '@Avalanche', dest: B, amount: token(30n) },
+  ]);
+});
+
+test('solver deposit redistribution (Deposit 100 -> A 70, B 30)', async t => {
+  const current = balances({ '<Deposit>': 100n, [A]: 0n, [B]: 0n });
+  const { steps } = await planRebalanceFlow({
+    mode: 'cheapest',
+    network: TEST_NETWORK,
+    current,
+    target: { '<Deposit>': ZERO, [A]: token(70n), [B]: token(30n) },
+    brand: TOK_BRAND,
+  });
+  t.deepEqual(steps, [
+    { src: '<Deposit>', dest: '@agoric', amount: token(100n) },
+    // TODO dckc should this  go through +agoric?
+    // { src: '+agoric', dest: '@agoric', amount: token(100n) },
+    { src: '@agoric', dest: '@noble', amount: token(100n) },
+    { src: '@noble', dest: '@Arbitrum', amount: token(70n) },
+    { src: '@noble', dest: '@Avalanche', amount: token(30n) },
+    { src: '@Arbitrum', dest: A, amount: token(70n) },
+    { src: '@Avalanche', dest: B, amount: token(30n) },
+  ]);
+});
+
+test('solver withdraw to cash (A 50 + B 30 -> Cash)', async t => {
+  const current = balances({ [A]: 50n, [B]: 30n, '<Cash>': 0n });
+  const { steps } = await planRebalanceFlow({
+    mode: 'cheapest',
+    network: TEST_NETWORK,
+    current,
+    target: { [A]: ZERO, [B]: ZERO, '<Cash>': token(80n) },
+    brand: TOK_BRAND,
+  });
+  t.deepEqual(steps, [
+    { src: A, dest: '@Arbitrum', amount: token(50n) },
+    { src: '@Arbitrum', dest: '@noble', amount: token(50n) },
+    { src: B, dest: '@Avalanche', amount: token(30n) },
+    { src: '@Avalanche', dest: '@noble', amount: token(30n) },
+    { src: '@noble', dest: '@agoric', amount: token(80n) },
+    { src: '@agoric', dest: '<Cash>', amount: token(80n) },
+  ]);
+});
+
+test('solver hub balances into pools (hubs supply -> pool targets)', async t => {
+  const current = balances({
+    [A]: 20n,
+    [B]: 10n,
+    [C]: 0n,
+    '@Arbitrum': 30n,
+    '@Avalanche': 20n,
+    '@noble': 20n,
+  });
+  const { steps } = await planRebalanceFlow({
+    mode: 'cheapest',
+    network: TEST_NETWORK,
+    current,
+    target: {
+      [A]: token(50n),
+      [B]: token(30n),
+      [C]: token(20n),
+      '@Arbitrum': ZERO,
```

Comment:
Interface question: can we omit targets with ZEROs?

Reply:
No. If there is not a target, that means "leave it the same" rather than "use those funds for the operation".